bokeh/palettes.py

```diff
@@ -1605,13 +1614,25 @@ def varying_alpha_palette(color: str, n: int | None = None, start_alpha: int = 0
     if not (0 <= end_alpha <= 255):
         raise ValueError(f"end_alpha {end_alpha} must be in the range 0 to 255")
 
+    # Take a copy of RGB color as do not want to alter named colors
+    rgba = NamedColor.from_string(color).copy()
+
+    if rgba.a < 1.0:
+        start_alpha = round(start_alpha*rgba.a)
+        end_alpha = round(end_alpha*rgba.a)
+
     if not n:
```

Comment:
When dealing with `None`, it should always be `if n is None`, and it should make mypy happy. However, if your intent was to cover `n == 0` as well, then do it explicitly.

Reply:
I'm also not in favour of treating numbers as booleans in general, thus such code is not permitted in bokehjs.

Reply:
`if not n` changed to `if n is None or n < 1`